src/db_adapters/access_keys.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::{ExpressionMethods, PgConnection, QueryDsl};
+use tokio_diesel::AsyncRunQueryDsl;
+use tracing::error;
+
+use near_indexer::near_primitives;
+
+use crate::models;
+use crate::schema;
+use diesel::pg::upsert::excluded;
+
+pub(crate) async fn handle_access_keys(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    outcomes: &near_indexer::ExecutionOutcomesWithReceipts,
+) {
+    let successful_receipts = outcomes
+        .values()
+        .filter(|outcome_with_receipt| {
+            match outcome_with_receipt.execution_outcome.outcome.status {
+                near_primitives::views::ExecutionStatusView::SuccessValue(_)
+                | near_primitives::views::ExecutionStatusView::SuccessReceiptId(_) => true,
+                _ => false,
+            }
+        })
+        .filter_map(|outcome_with_receipt| outcome_with_receipt.receipt.as_ref());
+
+    let mut access_keys = HashMap::<(String, String), models::access_keys::AccessKey>::new();
+
+    for receipt in successful_receipts {
+        if let near_primitives::views::ReceiptEnumView::Action { actions, .. } = &receipt.receipt {
+            for action in actions {
+                match action {
+                    near_primitives::views::ActionView::AddKey {
+                        public_key,
+                        access_key,
+                    } => {
+                        if let Some(existing_record) = access_keys
+                            .get_mut(&(public_key.to_string(), receipt.receiver_id.to_string()))
+                        {
+                            existing_record.created_by_receipt_id =
+                                Some(receipt.receipt_id.to_string());
+                            existing_record.deleted_by_receipt_id = None;
+                        } else {
+                            access_keys.insert(
+                                (public_key.to_string(), receipt.receiver_id.to_string()),
+                                models::access_keys::AccessKey::from_action_view(
+                                    public_key,
+                                    &receipt.receiver_id,
+                                    access_key,
+                                    &receipt.receipt_id,
+                                ),
+                            );
+                        }
```

Comment:
We don't want to reuse any of the previous fields if we see AddKey (imagine a case of AddKey FullAccess, DeleteKey, AddKey FunctionCall):

```suggestion
                        access_keys
                            .insert(
                                (public_key.to_string(), receipt.receiver_id.to_string()),
                                models::access_keys::AccessKey::from_action_view(
                                    public_key,
                                    &receipt.receiver_id,
                                    access_key,
                                    &receipt.receipt_id,
                                )
                            )
```